Rules/CTF/Scripts/CTF_GiveSpawnItems.as

```diff
@@ -4,99 +4,51 @@
 #include "CTF_Structs.as";
 
 const u32 materials_wait = 20; //seconds between free mats
-const u32 materials_wait_warmup = 40; //seconds between free mats
+const u32 materials_wait_warmup = 30; //seconds between free mats
 
 //property
-const string SPAWN_ITEMS_TIMER = "CTF SpawnItems:";
+const string SPAWN_ITEMS_TIMER_BUILDER = "CTF SpawnItems Builder:";
+const string SPAWN_ITEMS_TIMER_ARCHER  = "CTF SpawnItems Archer:";
 
 string base_name() { return "tent"; }
 
 bool SetMaterials(CBlob@ blob,  const string &in name, const int quantity)
 {
 	CInventory@ inv = blob.getInventory();
-
+	
 	//avoid over-stacking arrows
 	if (name == "mat_arrows")
 	{
 		inv.server_RemoveItems(name, quantity);
 	}
-
+	
 	CBlob@ mat = server_CreateBlobNoInit(name);
-
+	
 	if (mat !is null)
 	{
 		mat.Tag('custom quantity');
 		mat.Init();
-
+		
 		mat.server_SetQuantity(quantity);
-
+		
 		if (not blob.server_PutInInventory(mat))
 		{
 			mat.setPosition(blob.getPosition());
 		}
 	}
-
+	
 	return true;
 }
 
-bool GiveSpawnResources(CRules@ this, CBlob@ blob, CPlayer@ player, CTFPlayerInfo@ info)
-{
-	bool ret = false;
-
-	if (blob.getName() == "builder")
-	{
-		if (this.isWarmup())
-		{
-			ret = SetMaterials(blob, "mat_wood", 300) || ret;
-			ret = SetMaterials(blob, "mat_stone", 100) || ret;
-
-		}
-		else
-		{
-			ret = SetMaterials(blob, "mat_wood", 100) || ret;
-			ret = SetMaterials(blob, "mat_stone", 30) || ret;
-		}
-
-		if (ret)
-		{
-			info.items_collected |= ItemFlag::Builder;
-		}
-	}
-	else if (blob.getName() == "archer")
-	{
-		ret = SetMaterials(blob, "mat_arrows", 30) || ret;
-
-		if (ret)
-		{
-			info.items_collected |= ItemFlag::Archer;
-		}
-	}
-	else if (blob.getName() == "knight")
-	{
-		if (ret)
-		{
-			info.items_collected |= ItemFlag::Knight;
-		}
-	}
-
-	return ret;
-}
-
 //when the player is set, give materials if possible
 void onSetPlayer(CRules@ this, CBlob@ blob, CPlayer@ player)
 {
-	if (!getNet().isServer())
-		return;
-
-	if (blob !is null && player !is null)
-	{
-		RulesCore@ core;
-		this.get("core", @core);
-		if (core !is null)
-		{
-			doGiveSpawnMats(this, player, blob, core);
-		}
-	}
+	if (!getNet().isServer()) return;
```

Comment:
isServer() instead of getNet().isServer()

Reply:
fixed